### ISSUE TITLE
Error in inherits(x, "xml_document") 

### ISSUE THREAD

**Issue by seanchrismurphy** [jennybc/googlesheets#143]
When using edit_cells, I'm getting various errors of the form:

Error in inherits(x, "xml_document"): object '.' not found.

Specifically, this call throws an error:
sheet <- gs_title('Sheet 1')
edit_cells(sheet, input = 'Hello world', anchor = 'G1')

**Comment by jennybc** [comment]
Can you tell me exactly which version you've got? You can learn that from `devtools::session_info()` and sending me the line re: `googlesheets`. Just want to make sure I understand which commit you've installed from. In fact, it would also be interesting see your versions for `dplyr` and `xml2` while you're at it.

**Comment by jennybc** [comment]
Also the version of `cellranger`.

**Comment by seanchrismurphy** [comment]
Here is the output for seemingly-relevant packages from session_info() in a session where I still get the error:

cellranger   * 0.1.0      2015-04-22 CRAN (R 3.1.3)
dplyr        * 0.4.1.9000 2015-04-26 Github (hadley/dplyr@c1d82aa)
XML          * 3.98-1     2015-04-28 Github (omegahat/XML@4d95a6e)           
xml2         * 0.1.0      2015-04-20 CRAN (R 3.1.3) 
googlesheets   0.0.0.9001 2015-05-27 Github (jennybc/googlesheets@dd16645)
RCurl        * 1.95-4.6   2015-04-24 CRAN (R 3.1.3)

**Comment by jennybc** [comment]
A lot has changed since May 27, so before we do anything else, I ask that you reinstall `googlesheets` and `cellranger`, at the least:

```
install.packages(`cellranger`)
devtools::install_github("jennybc/googlesheets")
```

Those 2 packages are under my control and have definitely updated since you last installed.

I have no specific reason to believe you need to update anything else. However, I note that more recent versions of `dplyr`, `XML`, and `xml2` are available on CRAN as well. Unless you object, you may want to update these too.

Then let's see if the error persists.

Thanks.

**Comment by seanchrismurphy** [comment]
I've just updated all of those packages to the most recent available versions (and restarted R) and the error persists. 

The content in my 'input' call is of the form: "=IMAGE(\"http://i.imgur.com/xyz.jpg\", 1)"

Here is the current session_info output:

![screen shot 2015-06-23 at 3 09 01 pm](https://cloud.githubusercontent.com/assets/6520558/8299346/5f50fa5e-19ba-11e5-8c76-fd24359db30c.png)

**Comment by jennybc** [comment]
OK I will see if I can figure out what's problematic about this specific type of input.

**Comment by seanchrismurphy** [comment]
Sorry, I forgot but just retested it - I still get this error with 'Hello World'

**Comment by jennybc** [comment]
The fact you get the error with 'Hello World' is troubling. I sat down to make a little example from your snippet and noticed you say:

> edit_cells(sheet, input = 'Hello world', anchor = 'G1')

but `edit_cells()` hasn't been in `googlesheets` for a long time. So either you meant to say `gs_edit_cells()` or you are somehow still using a very old version.

Anyhow, I cooked up some code along your lines and it works for me, including the image formula insertion (I did have to resize the cell manually in the browser :)):

```
library(googlesheets)
ss <- gs_new("for_sean")
ss <- gs_edit_cells(ss, input = 'Hello world', anchor = 'B1')
gs_read_cellfeed(ss, range = 'B1')
ss <- gs_edit_cells(ss, input = "=IMAGE(\"http://i.imgur.com/lrg4uy5.jpg\", 2)",
                    anchor = 'B3')
```

```
> ss <- gs_new("for_sean")
Sheet "for_sean" created in Google Drive.
Worksheet dimensions: 1000 x 26.
> ss <- gs_edit_cells(ss, input = 'Hello world', anchor = 'B1')
Range affected by the update: "B1:B1"
Worksheet "Sheet1" successfully updated with 1 new value(s).
> gs_read_cellfeed(ss, range = 'B1')
Accessing worksheet titled "Sheet1"
Source: local data frame [1 x 5]

  cell cell_alt row col   cell_text
1   B1     R1C2   1   2 Hello world
> ss <- gs_edit_cells(ss, input = "=IMAGE(\"http://i.imgur.com/lrg4uy5.jpg\", 2)",
+ anchor = 'B3')
Range affected by the update: "B3:B3"
Worksheet "Sheet1" successfully updated with 1 new value(s).
```

and I attach a screenshot. Are you 100% sure you're using this new version of `googlesheets` you've clearly installed? Could you somehow be loading an older version due to multiple libraries?

![for_sean](https://cloud.githubusercontent.com/assets/599454/8314130/54cff51c-199c-11e5-82c3-9658e939d10b.png)

**Comment by seanchrismurphy** [comment]
I had been using edit_cells, but switched to gs_edit_cells after updating.
I'll try to run the above, though the one difference is a) I'm copying my
spreadsheets from a source sheet before editing and b) the cells are
resized before the copy (in a source spreadsheet)

**Comment by jennybc** [comment]
I don't think the fact that your source sheet comes from a copy should matter, nor should the size matter. But I could be wrong. If the above does not work, I'd ask that you give me read access to a representative sheet so I can get a better understanding of what you're seeing.

**Comment by seanchrismurphy** [comment]
So I ran the code you sent through and got the same error. I then downloaded and installed R 3.2.1., giving me a clean slate, and re-installed the latest versions of all relevant packages. The error now no longer appears. I suppose there must be an interaction with some other package, unless the latest version of R fixed the issue. I will report back should the issue return after re-installing more packages, but I guess the issue can be considered closed for now.
